dashboard/lib/certificate_image.rb

```diff
@@ -161,25 +161,125 @@ def self.create_course_certificate_image(name, course = nil, donor_name = nil, c
       vertical_offset = course == '20-hour' ? -125 : -120
       image = create_certificate_image2(path, name, y: vertical_offset)
       donor_text_y_offset = 447
-    elsif template_file == 'self_paced_pl_certificate.png'
-      image = Magick::Image.read(path).first
-      apply_text(image, name, 62, 'Helvetica bold', 'rgb(118,101,160)', 0, -248, CERT_NAME_AREA_WIDTH, 70)
-      course_title_width = 1400
-      course_title_height = 171
-      apply_text(image, course_title, 62, 'Helvetica bold', 'rgb(29,173,186)', 0, 0, course_title_width, course_title_height)
-
-      course_version = CurriculumHelper.find_matching_course_version(course)
-      total_hours = (course_version.content_root.duration_in_minutes / 60).floor
-      apply_text(image, total_hours.to_s, 30, 'Times bold', 'rgb(87,87,87)', -248, 124, 80, 30)
-      donor_text_y_offset = 611
-    else # all other courses use a certificate image where the course name is also blank
+    else
+      unit_or_unit_group = CurriculumHelper.find_matching_unit_or_unit_group(course)
       image = Magick::Image.read(path).first
-      apply_text(image, name, 75, 'Helvetica bold', 'rgb(118,101,160)', 0, -135, CERT_NAME_AREA_WIDTH, CERT_NAME_AREA_HEIGHT)
-      # The area in pixels which will display the course title.
-      course_title_width = 1000
-      course_title_height = 60
-      apply_text(image, course_title, 47, 'Helvetica bold', 'rgb(29,173,186)', 0, 15, course_title_width, course_title_height)
-      donor_text_y_offset = 447
+
+      text_constants = {
```

Comment:
I also wonder if these constants may have a better spot than here 🤔

Reply:
This seems like they could live in some sort of certificates helper file, but that's definitely a non-blocking (and preference, so feel free to disregard) nit!